### PR TITLE
refactor(Core/Misc): cos to std::cos

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  cos to std::cos

## Tests Performed:

- builds

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Object/ObjectPosSelector.h [hunk 1]
```diff
@@ -136,7 +136,7 @@ struct ObjectPosSelector
     UsedPosList::value_type const* nextUsedPos(UsedPosType uptype);
 
     // angle from used pos to next possible free pos
-    float GetAngle(UsedPos const& usedPos) const { return acos(m_dist / (usedPos.dist + usedPos.size + m_size)); }
+    float GetAngle(UsedPos const& usedPos) const { return std::cos(m_dist / (usedPos.dist + usedPos.size + m_size)); }
```
Comment (Nefertumm) [inline src/server/game/Entities/Object/ObjectPosSelector.h:139]:
```suggestion
    float GetAngle(UsedPos const& usedPos) const { return std::acos(m_dist / (usedPos.dist + usedPos.size + m_size)); }
```

File: src/server/game/Entities/Transport/Transport.cpp [hunk 2]
```diff
@@ -877,8 +877,8 @@ void StaticTransport::RelocateToProgress(uint32 progress)
         // pussywizard: PathRotation in db is only simple orientation rotation, so don't use sophisticated and not working code
         // reminder: WorldRotation only influences model rotation, not the path
         float sign = GetFloatValue(GAMEOBJECT_PARENTROTATION + 2) >= 0.0f ? 1.0f : -1.0f;
-        float pathRotAngle = sign * 2.0f * acos(GetFloatValue(GAMEOBJECT_PARENTROTATION + 3));
-        float cs = cos(pathRotAngle), sn = std::sin(pathRotAngle);
+        float pathRotAngle = sign * 2.0f * std::cos(GetFloatValue(GAMEOBJECT_PARENTROTATION + 3));
```
Comment (Nefertumm) [inline src/server/game/Entities/Transport/Transport.cpp:880]:
```suggestion
        float pathRotAngle = sign * 2.0f * std::acos(GetFloatValue(GAMEOBJECT_PARENTROTATION + 3));
```

File: src/server/game/Entities/Transport/Transport.cpp [hunk 3]
```diff
@@ -893,9 +893,9 @@ void StaticTransport::RelocateToProgress(uint32 progress)
         float percRot;
         m_goValue.Transport.AnimationInfo->GetAnimRotation(progress, currRot, nextRot, percRot);
         float signCurr = currRot.z >= 0.0f ? 1.0f : -1.0f;
-        float oriRotAngleCurr = signCurr * 2.0f * acos(currRot.w);
+        float oriRotAngleCurr = signCurr * 2.0f * std::cos(currRot.w);
```
Comment (Nefertumm) [inline src/server/game/Entities/Transport/Transport.cpp:896]:
```suggestion
        float oriRotAngleCurr = signCurr * 2.0f * std::acos(currRot.w);
```

File: src/server/game/Entities/Transport/Transport.cpp [hunk 4]
```diff
@@ -893,9 +893,9 @@ void StaticTransport::RelocateToProgress(uint32 progress)
...
         float signNext = nextRot.z >= 0.0f ? 1.0f : -1.0f;
-        float oriRotAngleNext = signNext * 2.0f * acos(nextRot.w);
+        float oriRotAngleNext = signNext * 2.0f * std::cos(nextRot.w);
```
Comment (Nefertumm) [inline src/server/game/Entities/Transport/Transport.cpp:898]:
```suggestion
        float oriRotAngleNext = signNext * 2.0f * std::acos(nextRot.w);
```

File: src/server/game/Entities/Object/ObjectPosSelector.cpp [hunk 5]
```diff
@@ -20,7 +20,7 @@
 ObjectPosSelector::ObjectPosSelector(float x, float y, float size, float dist)
     : m_center_x(x), m_center_y(y), m_size(size), m_dist(dist)
 {
-    m_anglestep = acos(m_dist / (m_dist + 2 * m_size));
+    m_anglestep = std::cos(m_dist / (m_dist + 2 * m_size));
```
Comment (Nefertumm) [inline src/server/game/Entities/Object/ObjectPosSelector.cpp:23]:
```suggestion
    m_anglestep = std::acos(m_dist / (m_dist + 2 * m_size));
```

**Review by Nefertumm — CHANGES_REQUESTED** [review]
Acos is not the same as cos.

File: src/server/game/Entities/Object/ObjectPosSelector.h (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Entities/Object/ObjectPosSelector.h:139]: that's odd as fuck, cuz I ran auto script for it to check explicitly for "cos" replace to "std::cos", where O.O

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Kitzunu** [comment]
Will just redo when I get time

**Pull request closed without merging**